### PR TITLE
Fix `target_vendor` in QNX Neutrino targets

### PR DESCRIPTION
**Pull request by madsmtm**
The `x86_64-pc-nto-qnx710` and `i586-pc-nto-qnx700` targets have `pc` in their target triple names, but the vendor was set to the default `"unknown"`.

CC target maintainers @flba-eb, @gh-tr, @jonathanpallant and @japaric

### CODE REVIEW

**Comment by workingjubilee** [comment]
not clear to me why the `target_vendor` isn't `qnx`, tbh.

**Comment by madsmtm** [comment]
The Aarch64 targets are named `aarch64-unknown-nto-qnx700` and `aarch64-unknown-nto-qnx710`, so may just be historical?

**Comment by jonathanpallant** [comment]
I don't know - what checks for this property?

**Review by wesleywiser — APPROVED** [review]
I don't really have an opinion so I'm happy to merge if any of the target maintainers agree we should do this!

**Comment by jonathanpallant** [comment]
I don't know what immediate value it adds, but it doesn't seem to hurt anything and I guess it's better to be consistent.

**Comment by jonathanpallant** [comment]
> not clear to me why the `target_vendor` isn't `qnx`, tbh.

My understanding is that `i386-pc-foo` specifically refers to an IBM PC compatible, as opposed to any other kind of machine you could build using an Intel x86 processor. It's "vendor" as in `sun`, `ibm`, or `apple`. Except there's no one vendor for PCs because it's an open standard, so it's just `pc`. That convention has then stuck.

**Comment by workingjubilee** [comment]
Ah, fair enough then!

> I don't really have an opinion so I'm happy to merge if any of the target maintainers agree we should do this!

> I don't know what immediate value it adds, but it doesn't seem to hurt anything and I guess it's better to be consistent. +1 from me.

`assert!(maintainers.iter().any(|maintainer| maintainer.agrees())`

@bors r=wesleywiser

**Pull request merged**
